src/main/java/homeWork_19/filtrowanieIPrzetwarzanie/java/FilteringMachine.java

```diff
@@ -0,0 +1,33 @@
+package homeWork_19.filtrowanieIPrzetwarzanie.java;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class FilteringMachine {
+
+	public List<Integer> filterOutNotEvenNumbers(List<Integer> numberList) {
+		return numberList.stream()
+				.filter(number -> number % 2 == 0)
+				.collect(Collectors.toList());
+	}
+
+	public List<Integer> filterOutLowerNumbersThan20(List<Integer> numberList) {
+		return numberList.stream()
+				.filter(number -> number > 20)
+				.collect(Collectors.toList());
+	}
+
+	public List<Book> convertToBooks(List<String> titles) {
+		return titles.stream()
+				.map(Book::new)
+				.collect(Collectors.toList());
+	}
+
+	// Metoda powinna przekształcić tytuły na książki i zwrócić tylko te które rozpoczynają się od słowa "Gra"
+	public List<Book> convertToBooksAndReturnOnlyStartingWithGra(List<String> titles) {
+		return titles.stream()
+				.map(Book::new)
+				.filter(title -> title.getTitle().startsWith("Gra"))
```

Comment:
Tutaj lepiej by było najpierwsz przefiltrować tytuły a potem stworzyć książki za pomocą metody map. Jest to istotne bo nie zamienisz od razu wszystkich książek na obiekty, tylko te które rzeczywiście tymi książkami powinny być. Taka oszczędność pamięci troche :)